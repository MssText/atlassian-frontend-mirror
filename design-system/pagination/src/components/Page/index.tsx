import React, { Component } from 'react';

import {
  Appearance,
  CustomThemeButton,
  CustomThemeButtonProps,
} from '@atlaskit/button';

type Diff<T, U> = T extends U ? never : T;

type PagePropsType = Diff<
  CustomThemeButtonProps,
  {
    appearance?: Appearance;
    autoFocus: boolean;
    isDisabled: boolean;
    isLoading: boolean;
    spacing: 'compact' | 'default' | 'none';
    shouldFitContainer: boolean;
    type: 'button' | 'submit';
  }
> & { page?: any };

export default class Page extends Component<PagePropsType> {
  render() {
    return <CustomThemeButton {...this.props} appearance="subtle" />;
  }
}
