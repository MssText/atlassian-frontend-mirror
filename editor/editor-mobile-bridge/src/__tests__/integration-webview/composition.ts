import { MobileTestCase } from '@atlaskit/webdriver-runner/runner';
import { SPECIAL_KEYS } from '@atlaskit/webdriver-runner/utils/mobile/wd-utils';
import Page from '@atlaskit/webdriver-runner/wd-app-wrapper';
import { loadEditor, getADFContent } from './_utils/afe-app-helpers';

MobileTestCase(
  'Composition: Typing on native keyboard',
  {},
  async (client, testName) => {
    const page = new Page(client);
    await loadEditor(page);

    const [, webViewContext] = await page.getContexts();

    // type on native keyboard
    // each letter sends composition start/update event, space key sends composition end
    if (page.isIOS()) {
      // iOS auto-capitalises start of the line, so turn that off
      // todo: make this not needed ED-8409
      await page.sendKeys([SPECIAL_KEYS.SHIFT]);
    }
    await page.sendKeys(['d', 'o', 'g', SPECIAL_KEYS.SPACE]);

    await page.switchContext(webViewContext);
    const adfContent = await getADFContent(page);

    expect(adfContent).toMatchCustomDocSnapshot(testName);
  },
);
