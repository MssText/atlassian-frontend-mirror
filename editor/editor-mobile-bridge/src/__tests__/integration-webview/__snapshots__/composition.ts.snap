// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Composition: Typing on native keyboard 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "text": "dog ",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "doc",
  "version": 1,
}
`;
